### ISSUE TITLE
[C++] Create fragment scanners for csv/parquet/orc/ipc

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#32566]
Once we have the basic scan node ready (with an initial implementation based on in-memory fragments) then we can add over the file-format versions.  We may also want to consider adding JSON support for datasets at this time.

**Reporter**: [Weston Pace](https://issues.apache.org/jira/browse/ARROW-17288) / @westonpace
**Assignee**: [Weston Pace](https://issues.apache.org/jira/browse/ARROW-17288) / @westonpace
#### PRs and other links:
- [GitHub Pull Request #14663](https://github.com/apache/arrow/pull/14663)

<sub>**Note**: *This issue was originally created as [ARROW-17288](https://issues.apache.org/jira/browse/ARROW-17288). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by raulcd** [comment]
@westonpace was this PR closing this ticket? https://github.com/apache/arrow/pull/14663, can we create any follow ups on a different ticket so I can assign this to the 11.0.0 milestone?

**Comment by westonpace** [comment]
No.  The ticket is for all the formats.  The PR only added support for one of them so I didn't close the ticket.  Perhaps unorthadox but I didn't want to split the JIRA issue into 4 similar issues.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 365 days. Please remove the stale label or comment below, or this issue will be closed in 14 days. If this improvement is still desired but has no current owner, please add the 'Status: needs champion' label.
